packages/cli/core/src/commands/ui/deploy.ts

```diff
@@ -0,0 +1,184 @@
+    pageId: Flags.string({
+      char: 'p',
+      description: 'The existing ID of the target Hosted search page.',
+      helpValue: '7944ff4a-9943-4999-a3f6-3e81a7f6fb0a',
+      required: false,
+    }),
+    config: Flags.string({
+      char: 'c',
+      description: 'The path to the deployment JSON configuration.',
+      helpValue: 'coveo.deploy.json',
+      default: 'coveo.deploy.json',
+      required: false,
+    }),
+  };
+
+  @Trackable()
+  @Preconditions(
+    IsAuthenticated([AuthenticationType.OAuth]),
+    HasNecessaryCoveoPrivileges(createSearchPagesPrivilege)
+  )
+  public async run() {
+    const {flags} = await this.parse(Deploy);
+    const deployConfigPath = flags.config;
+    const deployConfig: DeployConfig =
+      readJsonSync(deployConfigPath, {throws: false}) || {};
+    this.validateDeployConfigJson(deployConfigPath, deployConfig);
+    const hostedPage = this.buildHostedPage(deployConfig);
+
+    if (flags.pageId) {
+      this.updatePage({...hostedPage, id: flags.pageId});
+      return;
+    }
+
+    try {
+      await this.createPage(hostedPage);
+    } catch (error) {
+      /**
+       * TODO: handle Platform Client 400 error e.g.:
+       * {
+       *  statusCode: 400,
+       *  message: "A configuration with that name already exist.",
+       *  type: 'DuplicateConfigurationException'
+       * }
+       */
+      console.error(error);
+      super.catch(error);
+    }
+  }
+
+  private validateDeployConfigJson(
+    deployConfigPath: string,
+    config: DeployConfig
+  ) {
+    const validation = validate(config, DeployConfigSchema);
+    if (!validation.valid) {
+      throw new DeployConfigError(deployConfigPath, validation.errors);
+    }
+  }
+
+  private buildHostedPage({
+    dir,
+    htmlEntryFile,
+    name,
+    cssEntryFiles = [],
+    cssUrls = [],
+    javascriptEntryFiles = [],
+    javascriptUrls = [],
+  }: DeployConfig): New<HostedPage> {
+    ensureDirSync(dir);
+
+    return {
+      name,
+      html: readFileSync(join(dir, htmlEntryFile.path), 'utf-8'),
+      javascript: [
+        ...javascriptEntryFiles.map(({path, isModule}) => ({
+          inlineContent: readFileSync(join(dir, path), 'utf-8'),
+          isModule,
+        })),
+        ...javascriptUrls.map(({path, isModule}) => ({
+          url: path,
+          isModule,
+        })),
+      ],
+      css: [
+        ...cssEntryFiles.map(({path}) => ({
+          inlineContent: readFileSync(join(dir, path), 'utf-8'),
+        })),
+        ...cssUrls.map(({path}) => ({
+          url: path,
+        })),
+      ],
+    };
+  }
+
+  private async createClient() {
+    const authenticatedClient = new AuthenticatedClient();
+    return await authenticatedClient.getClient();
+  }
+
+  private async createPage(page: New<HostedPage>) {
+    this.log(`Creating new Hosted Page named "${page.name}".`);
+
+    const client = await this.createClient();
+    const response = await client.hostedPages.create(page);
```

Comment:
I think that would be a good place to use `startSpinner` since this is an async task:
example:
```
startSpinner('Creating hosted search page)
...
stopSpinner(`Hosted Page creation successful with id "${response.id}".`);
```